diff -ru nbsdsrc/src/usr.bin/chpass/Makefile usr.bin/chpass/Makefile
--- nbsdsrc/src/usr.bin/chpass/Makefile
+++ usr.bin/chpass/Makefile
@@ -8,11 +8,15 @@
 SRCS=	chpass.c edit.c field.c table.c util.c
 BINOWN=	root
 BINMODE=4555
-.PATH:	${NETBSDSRCDIR}/lib/libc/gen
+LIBC_DIR= nbsd_libc
+.PATH:	${NETBSDSRCDIR}/lib/${LIBC_DIR}/gen
 LINKS=	${BINDIR}/chpass ${BINDIR}/chfn ${BINDIR}/chpass ${BINDIR}/chsh
 MLINKS=	chpass.1 chfn.1 chpass.1 chsh.1
+CPPFLAGS+=-I${NETBSDSRCDIR}/lib/${LIBC_DIR}/include
 
-CPPFLAGS+=-I${NETBSDSRCDIR}/lib/libc/include
+.if defined(__MINIX)
+USE_YP=	no
+.endif
 
 .if (${USE_YP} != "no")
 SRCS+=	pw_yp.c
diff -ru nbsdsrc/src/usr.bin/chpass/edit.c usr.bin/chpass/edit.c
--- nbsdsrc/src/usr.bin/chpass/edit.c
+++ usr.bin/chpass/edit.c
@@ -151,8 +151,13 @@
 	int len, fd;
 	static char buf[LINE_MAX];
 
+#ifdef __minix
+	if ((fd = open(tempname, O_RDONLY)) == -1 ||
+	    (fp = fdopen(fd, "r")) == NULL)
+#else
 	if ((fd = open(tempname, O_RDONLY|O_NOFOLLOW)) == -1 ||
 	    (fp = fdopen(fd, "r")) == NULL)
+#endif
 		(*Pw_error)(tempname, 1, 1);
 	if (fstat(fd, &sb))
 		(*Pw_error)(tempname, 1, 1);
