diff -ru nbsdsrc/src/usr.sbin/user/Makefile usr.sbin/user/Makefile
--- nbsdsrc/src/usr.sbin/user/Makefile
+++ usr.sbin/user/Makefile
@@ -62,7 +62,7 @@
 	@echo "1. Adding new user"
 	@rm -f useradd
 	@ln -s user useradd
-	-./useradd -m -g=uid test1.1
+	-./useradd -m test1.1
 	@echo "2. Modifying new user"
 	-./${PROG} mod -l test1.2 test1.1
 	@echo "3. Deleting new user"
diff -ru nbsdsrc/src/usr.sbin/user/user.c usr.sbin/user/user.c
--- nbsdsrc/src/usr.sbin/user/user.c
+++ usr.sbin/user/user.c
@@ -152,8 +152,12 @@
 #endif
 
 #ifndef DEF_SKELDIR
+#ifdef __minix
+#define DEF_SKELDIR	"/usr/ast"
+#else
 #define DEF_SKELDIR	"/etc/skel"
 #endif
+#endif
 
 #ifndef DEF_SHELL
 #define DEF_SHELL	_PATH_BSHELL
@@ -386,7 +390,7 @@
 		warnx("Can't create group `%s': already exists", group);
 		return 0;
 	}
-	if ((from = fopen(_PATH_GROUP, "r")) == NULL) {
+	if ((from = fopen(_PATH_GROUP, "r+")) == NULL) {
 		warn("Can't create group `%s': can't open `%s'", name,
 		    _PATH_GROUP);
 		return 0;
@@ -450,7 +454,7 @@
 	int		fd;
 	int		cc;
 
-	if ((from = fopen(_PATH_GROUP, "r")) == NULL) {
+	if ((from = fopen(_PATH_GROUP, "r+")) == NULL) {
 		warn("Can't modify group `%s': can't open `%s'",
 		    group, _PATH_GROUP);
 		return 0;
@@ -573,7 +577,7 @@
 			}
 		}
 	}
-	if ((from = fopen(_PATH_GROUP, "r")) == NULL) {
+	if ((from = fopen(_PATH_GROUP, "r+")) == NULL) {
 		warn("Can't append group(s) for `%s': can't open `%s'",
 		    user, _PATH_GROUP);
 		return 0;
@@ -1088,7 +1092,7 @@
 		    login_name, up->u_class);
 	}
 #endif
-	if ((masterfd = open(_PATH_MASTERPASSWD, O_RDONLY)) < 0) {
+	if ((masterfd = open(_PATH_MASTERPASSWD, O_RDWR)) < 0) {
 		err(EXIT_FAILURE, "Can't add user `%s': can't open `%s'",
 		    login_name, _PATH_MASTERPASSWD);
 	}
@@ -1321,7 +1325,7 @@
 		    buf);
 		return 0;
 	}
-	if ((from = fopen(_PATH_GROUP, "r")) == NULL) {
+	if ((from = fopen(_PATH_GROUP, "r+")) == NULL) {
 		warn("Can't remove user `%s' from `%s': can't open `%s'",
 		    login_name, _PATH_GROUP, _PATH_GROUP);
 		return 0;
@@ -1450,7 +1454,7 @@
 	/* keep dir name in case we need it for '-m' */
 	homedir = pwp->pw_dir;
 
-	if ((masterfd = open(_PATH_MASTERPASSWD, O_RDONLY)) < 0) {
+	if ((masterfd = open(_PATH_MASTERPASSWD, O_RDWR)) < 0) {
 		err(EXIT_FAILURE, "Can't modify user `%s': can't open `%s'",
 		    login_name, _PATH_MASTERPASSWD);
 	}
