diff -ru nbsdsrc/src/usr.bin/newgrp/Makefile usr.bin/newgrp/Makefile
--- nbsdsrc/src/usr.bin/newgrp/Makefile
+++ usr.bin/newgrp/Makefile
@@ -3,6 +3,10 @@
 
 .include <bsd.own.mk>
 
+.if defined(__MINIX)
+USE_KERBEROS= no
+.endif
+
 PROG=	newgrp
 SRCS=	newgrp.c grutil.c
 BINOWN=	root
@@ -10,7 +14,9 @@
 
 CPPFLAGS+=-DGRUTIL_ACCEPT_GROUP_NUMBERS
 CPPFLAGS+=-DGRUTIL_ALLOW_GROUP_ERRORS	# for POSIX.1 compliance
+.if !defined(__MINIX)
 CPPFLAGS+=-DLOGIN_CAP
+.endif
 
 DPADD+=	${LIBCRYPT} ${LIBUTIL}
 LDADD+= -lcrypt -lutil
