diff -ru nbsdsrc/src/usr.sbin/pwd_mkdb/Makefile usr.sbin/pwd_mkdb/Makefile
--- nbsdsrc/src/usr.sbin/pwd_mkdb/Makefile
+++ usr.sbin/pwd_mkdb/Makefile
@@ -5,10 +5,14 @@
 
 PROG=	pwd_mkdb
 MAN=	pwd_mkdb.8
+.if defined(__MINIX)
+CPPFLAGS+=	-I${NETBSDSRCDIR}/lib/nbsd_libc/include
+.else
 CPPFLAGS+=	-I${NETBSDSRCDIR}/lib/libc/include
+.endif
 
 .ifndef HOSTPROG
-LDADD+=	-lutil
+LDADD+=	-lc -lutil
 DPADD+=	${LIBUTIL}
 .endif
 
